### ISSUE TITLE
[Wildcard Variables] `empty_catches` support

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4974]
Today if you want to catch and ignore an exception you can silence `empty_catches` by adding a comment to the catch clause,

```dart
try {
  ...
} catch(e) {
  // ignored, really.
}
```

With wildcards, I wonder if this shouldn't be sufficient to silence the lint?

```dart
try {
  ...
} catch(_) { }
```

**EDIT:** this is already how it's implemented but there's an open question about whether we want to start reporting on the non-wildcard (`catch(e)`) case.

**Comment by srawlins** [comment]
According to the [details section](https://dart.dev/tools/linter-rules/empty_catches#details), that already is good enough?

**Comment by lrhn** [comment]
The goal of the lint is to avoid *accidentally* not handling an error, forgetting to put something in the catch clause.
I think a `catch (_)` can be considered as a hint that it's deliberate. It's not a guarantee, but
```dart
} catch (e) {
  // Remember to handle the error!
}
```
is also a false negative.

A catch-all is always a code smell. There are places where it's needed, around untrusted self-contained code being run by a framework, but generally one shouldn't ignore errors. This lint is not enforcing that, it's just making sure you intended it, and `catch (_)` seems like a reasonable sign of intent.

**Comment by pq** [comment]
Thanks, @lrhn. 

> I think a catch (_) can be considered as a hint that it's deliberate.

I agree and has the advantage of a non-breaking change.

As for whether we should start reporting on `} catch (e) {` (even w/ a comment in the body), this will be more impacting (though easily remedied w/ `dart fix`).  Since it's a core lint, it'd be great to get some more input...

/fyi @goderbauer @mit-mit @dart-lang/language-team

**Comment by pq** [comment]
**Proposal A : don't lint `empty_catches` on `catch(_)` even without a comment in the body**

**Comment by pq** [comment]
**Proposal B : do lint `empty_catches` on `catch(e)` even with a comment in the body**

**Comment by pq** [comment]
Alternatively, we could nudge users to wildcards by flagging the `e` in 

```dart
try {
} catch(e) {
  // Unused.
}
```

as unused by producing an `UNUSED_CATCH_CLAUSE` diagnostic  (https://github.com/dart-lang/sdk/issues/55723).

**Comment by bwilkerson** [comment]
The `unused_catch_clause` diagnostic is only reported if there's an `on` clause. Removing the `catch` clause when there isn't an `on` clause would result in invalid code. Even if the user had
```dart
try {
  // ...
} on Object catch (_) {
}
```
the catch clause would still be unnecessary and should still be reported by `unused_catch_clause`.

**Comment by pq** [comment]
> The unused_catch_clause diagnostic is only reported if there's an on clause.

Oh, right!  I got the diagnostics confused and was thinking about the idea of beginning to report an `UNUSED_LOCAL_VARIABLE` diagnostic in all but the wildcard case.

That is:

```dart
try {
} catch(e) { // UNUSED_LOCAL_VARIABLE
}
```

but not:

```dart
try {
} catch(_) {
}
```

**Comment by bwilkerson** [comment]
Yes, if there's no `on` clause then I agree, we should report `unused_local_variable` and encourage users to use a wildcard.

**Comment by pq** [comment]
Closing in favor of conversation in https://github.com/dart-lang/sdk/issues/56595
